shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm

```diff
@@ -0,0 +1,119 @@
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.h"
+
+#include <OpenGL/gl.h>
+
+enum {
+  kFront = 0,
+  kBack = 1,
+  kBufferCount,
+};
+
+@interface FlutterSurfaceManager () {
+  CGSize surfaceSize;
+  CALayer* layer;
+  NSOpenGLContext* openGLContext;
+  uint32_t _frameBufferId[kBufferCount];
+  uint32_t _backingTexture[kBufferCount];
+  IOSurfaceRef _ioSurface[kBufferCount];
+}
+@end
+
+@implementation FlutterSurfaceManager
+
+- (instancetype)initWithLayer:(CALayer*)layer_ openGLContext:(NSOpenGLContext*)opengLContext_ {
+  if (self = [super init]) {
+    layer = layer_;
+    openGLContext = opengLContext_;
+
+    NSOpenGLContext* prev = [NSOpenGLContext currentContext];
+    [openGLContext makeCurrentContext];
+    glGenFramebuffers(2, _frameBufferId);
+    glGenTextures(2, _backingTexture);
+
+    glBindFramebuffer(GL_FRAMEBUFFER, _frameBufferId[0]);
+    glBindTexture(GL_TEXTURE_RECTANGLE_ARB, _backingTexture[0]);
+    glTexParameterf(GL_TEXTURE_RECTANGLE_ARB, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+    glTexParameterf(GL_TEXTURE_RECTANGLE_ARB, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
```

Comment:
It might also be worth doing:
```
glTexParameteri(GL_TEXTURE_RECTANGLE_ARB, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
glTexParameteri(GL_TEXTURE_RECTANGLE_ARB, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
```

To avoid having the default which is `GL_REPEAT`.

Reply:
I'm not sure this does anything give that we don't actually use these as textures anywhere, but I suppose it's not going to harm.